dd-sdk-android/src/main/kotlin/com/datadog/android/webview/internal/log/WebLogEventDateCorrector.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.webview.internal.log
+
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.log.internal.utils.buildLogDateFormat
+import java.lang.IllegalArgumentException
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+internal class WebLogEventDateCorrector(
+    private val timeProvider: TimeProvider,
+    private val simpleDateFormat: SimpleDateFormat =
+        buildLogDateFormat()
+) {
+
+    fun correctDate(date: String): String? {
+        return correctLogDate(date)
+    }
+
+    private fun correctLogDate(date: String): String? {
+        synchronized(simpleDateFormat) {
+            return try {
+                val time = simpleDateFormat.parse(date)?.time
+                if (time != null) {
+                    simpleDateFormat.format(time + timeProvider.getServerOffsetMillis())
+                } else {
+                    null
+                }
+            } catch (e: ParseException) {
+                sdkLogger.e(DATE_PARSING_ERROR_MESSAGE.format(Locale.US, date), e)
+                null
+            } catch (e: IllegalArgumentException) {
+                // this should not happen
```

Comment:
This comment is totally unnecessary. If it really should not happen, then remove the case entirely and add a `@Suppress`, if you think it might happen, then remove the comment. 
Also it'd be better to split the try catch as we don't know which call can throw `IllegalArgumentException`. I had to look to know that `parse()` can throw an NPE or a ParseException, and `format()` can throw an NPE and IllegalArgumentException

Reply:
Yes...based on our code it cannot happen as we are formatting a Long and it should not fail. Then again that method is marked with throw IAE because it accepts an `Object` and I said just in case I will add the `catch` block. I think I will still keep it but split it and remove the comment.